src/algorithms/reco/TransformBreitFrame.cc

```diff
@@ -0,0 +1,155 @@
+                                                                                         const edm4eic::InclusiveKinematicsCollection *kine,
+                                                                                         const edm4eic::ReconstructedParticleCollection *lab_collection) {
+    // Store the transformed particles
+    std::unique_ptr<edm4eic::ReconstructedParticleCollection> breit_collection { std::make_unique<edm4eic::ReconstructedParticleCollection>() };
+
+    // Beam momenta extracted from MCParticle
+    // This is the only place truth information is used!
+
+    // Get incoming electron beam
+    const auto ei_coll = find_first_beam_electron(mcpart);
+    if (ei_coll.size() == 0) {
+      m_log->debug("No beam electron found");
+      return breit_collection;
+    }
+    const PxPyPzEVector e_initial(
+      round_beam_four_momentum(
+        ei_coll[0].getMomentum(),
+        m_electron,
+        {-5.0, -10.0, -18.0},
+        0.0)
+      );
+
+    // Get incoming hadron beam
+    const auto pi_coll = find_first_beam_hadron(mcpart);
+    if (pi_coll.size() == 0) {
+      m_log->debug("No beam hadron found");
+      return breit_collection;
+    }
+    const PxPyPzEVector p_initial(
+      round_beam_four_momentum(
+        pi_coll[0].getMomentum(),
+        pi_coll[0].getPDG() == 2212 ? m_proton : m_neutron,
+        {41.0, 100.0, 275.0},
+        m_crossingAngle)
+      );
+
+    m_log->debug("electron energy, proton energy = {},{}",e_initial.E(),p_initial.E());
+
+    // Get the event kinematics, set up transform
+    if (kine->size() == 0) {
+      m_log->debug("No kinematics found");
+      return breit_collection;
+    }
+
+    const auto& evt_kin = kine->at(0);
+
+    const auto meas_x = evt_kin.getX();
+    const auto meas_Q2 = evt_kin.getQ2();
+
+    // Use relation to get reconstructed scattered electron
+    const auto ef_r = evt_kin.getScat();
+    const PxPyPzEVector e_final(ef_r.getMomentum().x, ef_r.getMomentum().y, ef_r.getMomentum().z, ef_r.getEnergy());
+
+    // Set up the transformation
+    const PxPyPzEVector virtual_photon = (e_initial - e_final);
+
+    m_log->debug("x, Q^2 = {},{}",meas_x,meas_Q2);
+
+    // Set up the transformation (boost) to the Breit frame
+    const auto P3 = p_initial.Vect();
+    const auto q3 = virtual_photon.Vect();
+    const ROOT::Math::Boost boost(-(2.0*meas_x*P3 + q3)*(1.0/(2.0*meas_x*p_initial.E() + virtual_photon.E())));
+    const auto breit = ROOT::Math::LorentzRotation(boost);
+
+    PxPyPzEVector p_initial_breit = (breit * p_initial);
+    PxPyPzEVector e_initial_breit = (breit * e_initial);
+    PxPyPzEVector e_final_breit = (breit * e_final);
+    PxPyPzEVector virtual_photon_breit = (breit * virtual_photon);
+
+    // Now rotate so the virtual photon momentum is all along the negative z-axis
+    const auto zhat =  -virtual_photon_breit.Vect().Unit();
+    const auto yhat = (e_initial_breit.Vect().Cross(e_final_breit.Vect())).Unit();
```

Comment:
I'm curious if we really need to rotate the interaction plane, probably unwanted for transverse spin measurements.

Reply:
To my understanding the Breit frame is *defined* with the rotation so that at LO the struck quark is along the -z-axis and the virtual photon along the  +z-axis.  

If you really wanted to do transverse spin measurements in the Breit frame you would transform a unit vector for the spin direction in the lab frame on and event-by-event basis.

Reply:
> To my understanding the Breit frame is _defined_ with the rotation so that at LO the struck quark is along the -z-axis and the virtual photon along the +z-axis.
> 
> If you really wanted to do transverse spin measurements in the Breit frame you would transform a unit vector for the spin direction in the lab frame on and event-by-event basis.

In your definition, you discuss just one direction $\hat{z}_{Breit}$, there would be a freedom of rotation around that axis. I was thinking along the lines of somehow defining $\hat{y}_{Breit}$ without referring to the reaction plane, but keeping whatever lab frame axes (proton spin info here) look after the boost. In any case, this is not something we can't fix later, just wanted to pick your brain about your motivation for doing it this way. It seems like it's not important to your analysis.

Reply:
Ah, I understand better now what you were getting at. There is no specific motivation for this choice, and I agree we can revisit this when necessary